api/src/main/java/uk/gov/dhsc/htbhf/claimant/repository/PaymentCycleEndDate.java

```diff
@@ -0,0 +1,23 @@
+package uk.gov.dhsc.htbhf.claimant.repository;
+
+import java.sql.Timestamp;
+import java.time.LocalDate;
+import java.util.UUID;
+
+/**
+ * Identifies a claim with a payment cycle due to expire on cycleEndDate.
+ */
+public interface PaymentCycleEndDate {
```

Comment:
Feels like this should have a better name as it refers to the end date itself rather than the PaymentCycle, so it should be something like `PaymentCycleWithEndDate` or with an appropriate adjective like `DefinedPaymentCycle`, `ClosedPaymentCycle`, `BoundedPaymentCycle` or `ActivePaymentCycle`. I've listed a few there as I'm not too happy with most of those, but you get the idea. It's slightly harder as it isn't used yet.

Reply:
How about ExpiringPaymentCycle?

Reply:
or ClosingPaymentCycle perhaps. Looking for more synonyms of 'ending'...

Reply:
I reckon either ClosingPaymentCycle or PaymentCycleWithEndDate.

Reply:
Naming is always hard isn't it! How about `ClosingPaymentCycle`, then should we need another field on here it won't seem wrong unlike `PaymentCycleWithEndDate`.